src/schema/v2/artwork/collectorSignals.ts

```diff
@@ -29,48 +27,267 @@ export const CollectorSignals: GraphQLFieldConfig<any, ResolverContext> = {
+        resolve: async (artwork, {}, ctx) => {
+          const isInSale = artwork.sale_ids?.length > 0
+
+          if (
+            !checkFeatureFlag("emerald_signals-auction-improvements", ctx) ||
+            !isInSale
+          ) {
+            return null
+          }
+
+          const activeLotData = await getActiveSaleArtwork(
+            {
+              artworkId: artwork.id,
+              saleIds: artwork.sale_ids,
+            },
+            ctx
+          )
+
+          if (activeLotData) {
+            return artwork.recent_saves_count
+          }
+
+          return null
+        },
       },
       registrationEndsAt: {
         type: GraphQLString,
         description: "Pending auction registration end time",
+        resolve: async (artwork, {}, ctx) => {
+          const isInSale = artwork.sale_ids?.length > 0
+
+          if (
+            !checkFeatureFlag("emerald_signals-auction-improvements", ctx) ||
+            !isInSale
+          ) {
+            return null
+          }
+
+          const activeLotData = await getActiveSaleArtwork(
+            {
+              artworkId: artwork.id,
+              saleIds: artwork.sale_ids,
+            },
+            ctx
+          )
+
+          if (activeLotData) {
+            const registrationEndAtDate =
+              activeLotData.sale.registration_ends_at &&
+              new Date(activeLotData.sale.registration_ends_at)
+
+            if (registrationEndAtDate && registrationEndAtDate > NOW) {
+              return registrationEndAtDate.toISOString()
+            }
+          }
+
+          return null
+        },
       },
       lotClosesAt: {
         type: GraphQLString,
         description: "Pending auction lot end time for bidding",
+        resolve: async (artwork, {}, ctx) => {
+          const isInSale = artwork.sale_ids?.length > 0
+
+          if (
+            !checkFeatureFlag("emerald_signals-auction-improvements", ctx) ||
+            !isInSale
+          ) {
+            return null
+          }
+
+          const activeLotData = await getActiveSaleArtwork(
+            {
+              artworkId: artwork.id,
+              saleIds: artwork.sale_ids,
+            },
+            ctx
+          )
+
+          if (activeLotData) {
+            const extendedBiddingEndAtDate =
+              activeLotData.saleArtwork.extended_bidding_end_at &&
+              new Date(activeLotData.saleArtwork.extended_bidding_end_at)
+
+            if (extendedBiddingEndAtDate && extendedBiddingEndAtDate > NOW) {
+              return extendedBiddingEndAtDate.toISOString()
+            }
+
+            const lotClosesAtDate =
+              activeLotData.saleArtwork.end_at &&
+              new Date(activeLotData.saleArtwork.end_at)
+
+            return lotClosesAtDate && lotClosesAtDate.toISOString()
+          }
+
+          return null
+        },
       },
       onlineBiddingExtended: {
         type: GraphQLBoolean,
```

Comment:
Same as above re: non-null, maybe consider `auction` group within signals that evaluates to `null` unless the work is in an auction and isn't closed yet.